course/src/Isometry.cc

```diff
@@ -92,5 +94,23 @@ const Vector3 Vector3::kUnitY(0.0, 1.0, 0.0);
 const Vector3 Vector3::kUnitZ(0.0, 0.0, 1.0);
 const Vector3 Vector3::kZero(0.0, 0.0, 0.0);
 
+// Matrix3
+
+const Matrix3 Matrix3::kIdentity({1.0, 0.0, 0.0}, {0.0, 1.0, 0.0}, {0.0, 0.0, 1.0});
+const Matrix3 Matrix3::kZero({0.0, 0.0, 0.0}, {0.0, 0.0, 0.0}, {0.0, 0.0, 0.0});
+
+bool Matrix3::operator==(const Matrix3& a) const {
+    auto it = values_.begin();
+    auto a_it = a.values_.begin();
+    for (uint8_t count = 0; count < values_.size(); ++count) {
```

Comment:
What happens if both matrices don't have the same number of elements? Code defensively.

Reply:
If I pass a Matrix3 as a parameter, shouldn't be always initialized?

Reply:
Now they are, but before this you were not checking the length of the initliazer list constructor.